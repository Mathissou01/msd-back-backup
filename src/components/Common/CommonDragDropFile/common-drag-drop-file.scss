@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.buttons" as buttons;
@use "src/styles/02-tools/tools.typography" as typo;

.c-CommonDragDropFile {
  position: relative;
  width: 100%;

  &__LabelDragDrop {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100%;
    padding: 15px 0;
    border: 2px dashed colors.$blue-5;
    border-radius: 1rem;
  }

  &__Drag {
    display: flex;
    flex-direction: column;
    gap: 18px;
    align-items: center;
  }

  &__LabelInfo {
    color: colors.$expert-blue;
    font-weight: 500;
    font-size: 20px;
  }
}
