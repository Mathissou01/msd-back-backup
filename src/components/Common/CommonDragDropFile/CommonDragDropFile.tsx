import React, { useRef } from "react";
import Image from "next/image";
import { TAcceptedMimeTypes } from "../../../lib/media";
import CommonButton from "../CommonButton/CommonButton";
import "./common-drag-drop-file.scss";

interface ICommonDragDropFileProps {
  handleDragOver: (event: { preventDefault: () => void }) => void;
  handleDrop: (event: React.DragEvent<HTMLFormElement>) => void;
  handleFileChange: (event: { target: HTMLInputElement }) => void;
  acceptedTypes?: Array<TAcceptedMimeTypes>;
}

export default function CommonDragDropFile({
  handleDragOver,
  handleDrop,
  handleFileChange,
}: ICommonDragDropFileProps) {
  /* Static Data */
  const labels = {
    labelInfosDragDrop: "Glissez-d√©posez une image dans cette zone ou",
    uploadFile: "Parcourir les fichiers",
  };
  const mediaImage = "/images/pictos/add-photo.svg";

  /* Local Data */
  const inputRef = useRef<HTMLInputElement>(null);

  return (
    <>
      <form
        className="c-CommonDragDropFile"
        onDragOver={handleDragOver}
        onDrop={(e) => handleDrop(e)}
        data-testid="common-drag-drop-file-form"
      >
        <input
          type="file"
          hidden
          id="input-file-upload"
          data-testid="common-drag-drop-file-input"
          multiple={true}
          ref={inputRef}
          onChange={handleFileChange}
        />
        <label
          className="c-CommonDragDropFile__LabelDragDrop"
          htmlFor="input-file-upload"
        >
          <div className="c-CommonDragDropFile__Drag">
            <Image src={mediaImage} alt="" width={50} height={50} />
            <span className="c-CommonDragDropFile__LabelInfo">
              {labels.labelInfosDragDrop}
            </span>
            <CommonButton
              type="button"
              style="primary"
              label={labels.uploadFile}
              onClick={() => inputRef.current?.click()}
            />
          </div>
        </label>
      </form>
    </>
  );
}
