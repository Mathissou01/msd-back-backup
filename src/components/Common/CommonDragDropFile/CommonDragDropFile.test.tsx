import { fireEvent, render, screen } from "@testing-library/react";
import CommonDragDropFile from "./CommonDragDropFile";

describe("CommonDragDropFile", () => {
  const handleDragOver = jest.fn();
  const handleFileChange = jest.fn();
  const handleDrop = jest.fn();

  it("renders component", () => {
    const { container } = render(
      <CommonDragDropFile
        handleDragOver={handleDragOver}
        handleFileChange={handleFileChange}
        handleDrop={handleDrop}
      />,
    );
    expect(container).toMatchSnapshot();
    expect(handleDragOver).toReturnTimes(0);
    expect(handleFileChange).toReturnTimes(0);
    expect(handleDrop).toReturnTimes(0);
  });

  it("handles drag and drop events", () => {
    render(
      <CommonDragDropFile
        handleDragOver={handleDragOver}
        handleFileChange={handleFileChange}
        handleDrop={handleDrop}
      />,
    );

    const form = screen.getByTestId("common-drag-drop-file-form");
    expect(form).toBeInTheDocument();
    fireEvent.drop(form);
    expect(handleDrop).toBeCalledTimes(1);
    fireEvent.dragOver(form);
    expect(handleDragOver).toBeCalledTimes(1);
  });

  it("changes input value", () => {
    render(
      <CommonDragDropFile
        handleDragOver={handleDragOver}
        handleFileChange={handleFileChange}
        handleDrop={handleDrop}
      />,
    );

    const input = screen.getByTestId("common-drag-drop-file-input");
    expect(input).toBeInTheDocument();
    fireEvent.change(input);
    expect(handleFileChange).toBeCalledTimes(1);
  });
});
