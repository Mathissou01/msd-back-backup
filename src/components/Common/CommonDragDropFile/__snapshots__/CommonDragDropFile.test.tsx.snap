// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonDragDropFile renders component 1`] = `
<div>
  <form
    class="c-CommonDragDropFile"
    data-testid="common-drag-drop-file-form"
  >
    <input
      data-testid="common-drag-drop-file-input"
      hidden=""
      id="input-file-upload"
      multiple=""
      type="file"
    />
    <label
      class="c-CommonDragDropFile__LabelDragDrop"
      for="input-file-upload"
    >
      <div
        class="c-CommonDragDropFile__Drag"
      >
        <span
          class="c-CommonDragDropFile__LabelInfo"
        >
          Glissez-d√©posez une image dans cette zone ou
        </span>
        <button
          class="c-CommonButton c-CommonButton_primary c-CommonButton_fontSmall"
          type="button"
        >
          Parcourir les fichiers
        </button>
      </div>
    </label>
  </form>
</div>
`;
